Docs: Update "Install OpenVINO Runtime on Windows from Archive File" page (#13332)
openvinotoolkit/openvino: docs/install_guides/installing-openvino-from-archive-windows.md

# Install OpenVINO™ Runtime on Windows from an Archive File {#openvino_docs_install_guides_installing_openvino_from_archive_windows}

> With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as sample code for running demos. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.

Reviewer: ```suggestion
With the OpenVINO™ 2022.2 release, you can download and use archive files to install OpenVINO Runtime. The archive files contain pre-built binaries and library files needed for OpenVINO Runtime, as well as code samples. This page provides instructions showing how to install OpenVINO Runtime using archive files. Check the [Release Notes](https://software.intel.com/en-us/articles/OpenVINO-RelNotes) for more information on updates in the 2022.2 release.
```

@sphinxdirective
.. tab:: Operating Systems

  * Microsoft Windows 10, 64-bit

>   * Microsoft Windows 11, 64-bit

Reviewer 2: can we also highlight architecture we are targeting? E.g. explicit x86_64
This is a general recommendation for all the instructions, because OpenVINO also supports ARM

.. tab:: Software

  * `Microsoft Visual Studio 2019 with MSBuild <http://visualstudio.microsoft.com/downloads/>`_
  * `CMake 3.14 or higher, 64-bit <https://cmake.org/download/>`_ (optional, only required for building sample applications)
  * `Python 3.6 - 3.10, 64-bit <https://www.python.org/downloads/windows/>`_

>      * Note that OpenVINO is gradually phasing out support for Python 3.6. Python 3.7 - 3.10 are recommended.

Reviewer: yeah, this is for master. I will change it to 3.9 when porting to 2022.2.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

1. Select and download the OpenVINO™ archive files from [Intel® Distribution of OpenVINO™ toolkit download page](https://software.intel.com/en-us/openvino-toolkit/choose-download).
   There are typically two files for you to download: 
1. Open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.
   ```sh
   w_openvino_toolkit_<operating system>_<release version>_<package ID>_x86_64.zip
   w_openvino_toolkit_<operating system>_<release version>_<package ID>_x86_64.zip.sha256
   ``` 
   where the `.sha256` file is used to verify the success of the download process.
   
2. Locate the downloaded files in your system. This document assumes the files are in your `Downloads` directory. 
   
3. Open a command prompt terminal window, and run the following command with the `.zip` file:
   ```sh

>    CertUtil -hashfile <archive name>.zip SHA256

Reviewer 2: do you think we can keep this section under a spoiler?

Reviewer: Just want to check, what is a spoiler?

Reviewer 2: Sorry, I meant collapsible section like here https://github.com/openvinotoolkit/openvino/blob/master/docs/OV_Runtime_UG/deployment/deployment-manager-tool.md#running-deployment-manager-in-interactive-mode

Reviewer: @ilya-lavrenov Thanks for the suggestion. I didn't manage to create the collapsible block, so to save some time investigating, I included this info in a note. Hope that's good with you. If you don't mind, can I merge this PR?

>    ren w_openvino_toolkit_windows_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713

Reviewer 2: CC @artanokhov @evgenytalanin-intel

> Congratulations, you finished installation! The C:\Program Files (x86)\Intel\openvino_2022 folder now contains the core components for OpenVINO™. When other pages in OpenVINO™ documentation refer to the `<INSTALL_DIR>` directory, this is the folder they're referring to. If you installed OpenVINO™ in a different location, make sure to use that instead.

Reviewer: ```suggestion
Congratulations, you finished the installation! The `C:\Program Files (x86)\Intel\openvino_2022` folder now contains the core components for OpenVINO. If you used a different folder in Step 1, you will find the `openvino_2022` folder there..
```

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

> You must update several environment variables before you can compile and run OpenVINO™ applications. Open the Command Prompt, and run the `setupvars.bat` batch file to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `C:\Program Files (x86)\Intel\openvino_2022`, use that location instead.

Reviewer: ```suggestion
You must update several environment variables before you can compile and run OpenVINO™ applications. Open the Command Prompt, and run the `setupvars.bat` batch file to temporarily set your environment variables. If your <INSTALL_DIR> is not `C:\Program Files (x86)\Intel\openvino_2022`, use the correct directory instead.
```

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

1. Open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.
   ```sh
   mkdir "C:\Program Files (x86)\Intel"
   cd "C:\Program Files (x86)\Intel"
   ```

>    > **NOTE**: C:\Program Files (x86)\Intel is the recommended folder path, but you may use a different path if desired or if you don't have administrator priveleges on your PC. If the C:\Program Files (x86)\Intel folder already exists, skip the `mkdir` command.

Reviewer: ```suggestion
   > **NOTE**: C:\Program Files (x86)\Intel is the recommended folder. You may also use a different path if desired or if you don't have administrator privileges on your computer.
```

> 2. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), unzip it, and rename the folder to `openvino_2022.2.0.7713` by issuing:

Reviewer: ```suggestion
2. Download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0), extract it to `C:\Program Files (x86)\Intel`, and rename the extracted folder to `openvino_2022.2.0.7713`.
   If you want to use command-lines for this step, run the following commands in the terminal window you opened:
```

   ```sh
   curl -L https://github.com/openvinotoolkit/openvino/releases/download/2022.2.0/w_openvino_toolkit_windows_2022.2.0.7713.af16ea1d79a_x86_64.zip --output openvino_2022.2.0.7713.zip
   tar -xf openvino_2022.2.0.7713.zip
   ren w_openvino_toolkit_windows_2022.2.0.7713.af16ea1d79a_x86_64 openvino_2022.2.0.7713
   ```

> 3. Create a symbolic link to the folder by issuing:

Reviewer: ```suggestion
3. For simplicity, it is useful to create a symbolic link. Browse to `C:\Program Files (x86)\Intel` and run the following command in a command prompt window:
```

   ```sh
   mklink /D openvino_2022 openvino_2022.2.0.7713
   ```

> 

Reviewer: ```suggestion
```

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

1. Open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.
   ```sh
   mkdir "C:\Program Files (x86)\Intel"
   cd "C:\Program Files (x86)\Intel"
   ```

> 

Reviewer: ```suggestion
```

  * `Python 3.6 - 3.10, 64-bit <https://www.python.org/downloads/windows/>`_
     * Note that OpenVINO is gradually phasing out support for Python 3.6. Python 3.7 - 3.10 are recommended.

  .. note::
     To install Microsoft Visual Studio 2019, follow the `Microsoft Visual Studio installation guide <https://docs.microsoft.com/en-us/visualstudio/install/install-visual-studio?view=vs-2019>`_. You can choose to download the Community version. During installation in the **Workloads** tab, choose **Desktop development with C++**.

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

> 

Reviewer: I don't think using command-line to create a new folder, download and extract the package is a user-friendly way on Windows. Taking the user as new and non tech-savvy, I think we should recommend the most common way on Windows. Command-line could be an alternative option.

> 1. Open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.

Reviewer: ```suggestion
1. Create an 'Intel' folder in the `C:\Program Files (x86)\` directory for OpenVINO and browse to it. 
    You can also do this via command-lines by opening a terminal window as administrator and running the following commands:
```

> 

Reviewer: ```suggestion
```

     * Note that OpenVINO is gradually phasing out support for Python 3.6. Python 3.7 - 3.10 are recommended.

  .. note::
     To install Microsoft Visual Studio 2019, follow the `Microsoft Visual Studio installation guide <https://docs.microsoft.com/en-us/visualstudio/install/install-visual-studio?view=vs-2019>`_. You can choose to download the Community version. During installation in the **Workloads** tab, choose **Desktop development with C++**.

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

1. Create an 'Intel' folder in the `C:\Program Files (x86)\` directory. Skip this step if the folder already exists.

>    You can also do this via command-lines. Open a new command prompt window as administrator by right-clicking **Command Prompt** from the Start menu and select **Run as administrator**, and then run the following commands:

Reviewer: ```suggestion
   You can also do this via command-lines. Open a new command prompt window as administrator by right-clicking **Command Prompt** from the Start menu and select **Run as administrator**, and then run the following command:
```

.. tab:: Software

>   * `Microsoft Visual Studio 2019 with MSBuild <https://visualstudio.microsoft.com/vs/older-downloads/>_ or `Microsoft Visual Studio 2022 <http://visualstudio.microsoft.com/downloads/>`_

Reviewer: ```suggestion
  * `Microsoft Visual Studio 2019 with MSBuild <https://visualstudio.microsoft.com/vs/older-downloads/>`_ or `Microsoft Visual Studio 2022 <http://visualstudio.microsoft.com/downloads/>`_
```

.. tab:: Software

>   * `Microsoft Visual Studio 2019 with MSBuild <http://visualstudio.microsoft.com/downloads/>`_

Reviewer 2: We can add something like "MSVC of version 2019 at least", because 2022 is also supported

>   * `Microsoft Visual Studio 2019 with MSBuild <http://visualstudio.microsoft.com/downloads/>`_

Reviewer: ```suggestion
  * `Microsoft Visual Studio 2019 with MSBuild <https://visualstudio.microsoft.com/vs/older-downloads/>_ or `Microsoft Visual Studio 2022 <http://visualstudio.microsoft.com/downloads/>`_
```
I guess we can mention them explicitly.

  * `CMake 3.14 or higher, 64-bit <https://cmake.org/download/>`_ (optional, only required for building sample applications)

>   * `Python 3.6 - 3.10, 64-bit <https://www.python.org/downloads/windows/>`_

Reviewer 2: python 3.10 is not supported

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

1. Create an `Intel` folder in the `C:\Program Files (x86)\` directory. Skip this step if the folder already exists.
   
   You can also do this via command-lines. Open a new command prompt window as administrator by right-clicking **Command Prompt** from the Start menu and select **Run as administrator**, and then run the following command:
   ```sh
   mkdir "C:\Program Files (x86)\Intel"
   ```
   > **NOTE**: `C:\Program Files (x86)\Intel` is the recommended folder. You may also use a different path if desired or if you don't have administrator privileges on your computer.

> 2. Download the **OpenVINO Runtime archive file for Windows** from the [OpenVINO releases page](https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0) to your local `Downloads` folder.

Reviewer: ```suggestion
2. Download the [OpenVINO Runtime archive file for Windows](https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/windows/) to your local `Downloads` folder.
```

  * For Python developers: `Python 3.6 - 3.9, 64-bit <https://www.python.org/downloads/windows/>`_
     * Note that OpenVINO is gradually phasing out support for Python 3.6. Python 3.7 - 3.9 are recommended.

  .. note::
     To install Microsoft Visual Studio 2019, follow the `Microsoft Visual Studio installation guide <https://docs.microsoft.com/en-us/visualstudio/install/install-visual-studio?view=vs-2019>`_. You can choose to download Community version. During installation in the **Workloads** tab, choose **Desktop development with C++**.

  .. note::
    You can either use `cmake<version>.msi` which is the installation wizard or `cmake<version>.zip` where you have to go into the `bin` folder and then manually add the path to environmental variables.
  
  .. important::
    When installing Python, make sure you click the option **Add Python 3.x to PATH** to `add Python <https://docs.python.org/3/using/windows.html#installation-steps>`_ to your `PATH` environment variable.

@endsphinxdirective

## Installing OpenVINO Runtime

### <a name="install-openvino"></a>Step 1: Download and Install OpenVINO Core Components

> First, open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.

Reviewer 3: Ah I see. For Linux we had the option to install in $HOME/intel too, as a regular user. This is also useful for users on restricted systems, like schools. But if we never offered that for Windows, then at least this is not a new limitation. I agree that for existing users it's nice if it is in the same directory. Having to run an administrator command prompt adds complexity though :-( Now I'm overthinking too :-) Let's stick to program files for now, unless someone else has a different opinion.

> First, open a command terminal as administrator by searching for "Command Terminal" in the Start menu, right clicking on it, and selecting "Run as administrator". Create a folder for OpenVINO and move into it by issuing the following commands.

Author: Okay, I'll bring it up at the meeting tomorrow so we can discuss it!

```sh
mkdir "C:\Program Files (x86)\Intel"
cd "C:\Program Files (x86)\Intel"
```

> **NOTE**: C:\Program Files (x86)\Intel is the recommended folder path, but you may use a different path if desired or if you don't have administrator priveleges on your PC. If the C:\Program Files (x86)\Intel folder already exists, skip the `mkdir` command.

Next, you'll download the OpenVINO Runtime 2022.2 archive file from the [OpenVINO archives](https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/windows) site. Issue the following command to download archive file, unzip it, and rename the folder to `openvino_2022.2.0.7713`:
```sh

> curl https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/windows/w_openvino_toolkit_windows_2022.2.0.7713.af16ea1d79a_x86_64.zip --output openvino_2022.2.0.7713.zip

Reviewer 2: should we point to github archives? They have download statistics

Author: Yes, we could do that. What is the link to the GitHub archives?

Reviewer 2: Please, find it here https://github.com/openvinotoolkit/openvino/releases/tag/2022.2.0

Author: Thanks Ilya! I changed it to use the GitHub link instead.

Congratulations, you finished installation! The C:\Program Files (x86)\Intel\openvino_2022 folder now contains the core components for OpenVINO™. When other pages in OpenVINO™ documentation refer to the `<INSTALL_DIR>` directory, this is the folder they're referring to. If you installed OpenVINO™ in a different location, make sure to use that instead.

### <a name="set-the-environment-variables"></a>Step 2: Configure the Environment

You must update several environment variables before you can compile and run OpenVINO™ applications. Open the Command Prompt, and run the `setupvars.bat` batch file to temporarily set your environment variables. Again, if you installed OpenVINO™ in a folder other than `C:\Program Files (x86)\Intel\openvino_2022`, use that location instead.

```sh
"C:\Program Files (x86)\Intel\openvino_2022\setupvars.bat"
```

> **Important**: The above command must be re-run every time a new Command Prompt window is opened.

> > **NOTE**: If you see an error indicating Python is not installed, Python may not be added to the PATH environment variable (as described [here](https://docs.python.org/3/using/windows.html#finding-the-python-executable)). Check your system environment variables, and add Python if necessary.

Reviewer 3: To me this reads as if Python is required for using OpenVINO, but it is not, it is only required for using the Python API.

Author: Per our email conversation, I created Issue #13411 to report this problem.

The environment variables are set. Next, you can download some additional tools.

### <a name="model-optimizer">Step 3 (Optional): Install Additional Components</a>

> OpenVINO Development Tools adds even more functionality to OpenVINO. It provides tools like Model Optimizer, Benchmark Tool, Post-Training Optimization Tool, and Open Model Zoo Downloader. If you install OpenVINO Development Tools, OpenVINO Runtime will also be installed as a dependency, so you don't need to install OpenVINO Runtime separately. 

Reviewer 3: This is only true for Python developers (who use CPU and GPU plugins). The plugin limitation is maybe out of scope for this document, but the Python part is important. 

For Python developers, should we be clearer upfront that PyPI already has everything they need?

Author: This one is confusing :( :( :( . Let me think for a bit.

Author: What if we change it to this?

"OpenVINO Development Tools is a set of utilities for working with OpenVINO and OpenVINO models. It provides tools like Model Optimizer, Benchmark Tool, Post-Training Optimization Tool, and Open Model Zoo Downloader. If you install OpenVINO Runtime using archive files, OpenVINO Development Tools must be installed separately.

See the [Install OpenVINO Development Tools](https://github.com/EdjeElectronics/openvino/blob/update-pypi-whats-next/docs/install_guides/installing-model-dev-tools.md) page for step-by-step installation instructions."

Reviewer 3: Both suggestions are great, thanks @EdjeElectronics ! Unfortunately PyPI does not include MYRIAD/HDDL at the moment. Maybe to the second suggestion add something about that PyPI includes both CPU and GPU plugin?

Author: Okay, I changed the page to add both suggestions.
